### DISCUSSION TITLE
Improve Rauc info command to verify the compatible field

### DISCUSSION THREAD

**Discussion by hatmaalla** [rauc/rauc#1062] [General]
Hello,
I use the rauc info(URL) command to check the integrity, authenticity and compatibility of the bundle with the hardware before downloading and installing it.
The idea behind this is to avoid unnecessary downloading if the bundle is not compatible with the target, especially the `compatible` field in `/etc/rauc/system.conf`.
The problem is that the rauc info command does not seem to check the `compatible` field with the one mentioned in the manifest.
So I have to manually read `/etc/rauc/system.conf` and compare it with the return of the rauc info command to decide whether to download or not.
Is it possible to improve the rauc info command so that it also checks for `compatible` and returns KO if it doesn't match the manifest?

**Comment by ejoerns** [comment]
Please have a look at RAUC streaming capabilities which don't require unnecessary downloads of unused bundle payload:

https://rauc.readthedocs.io/en/latest/advanced.html#http-streaming

**Reply by ejoerns** [reply]
The purpose of `rauc info` is primarily to inspect the bundle, not to test its compatibility with the target. It is more or less a 'host' tool.

**Reply by hatmaalla** [reply]
Thanks for the feedback,
The use case on my side is to ask the user for confirmation before starting the installation, in other words, I run `RAUC info`, if everything is OK (bundle content and compatibility), I inform the user that a new compatible version is available, if the user accepts, I start the installation.
Thus, any errors that occur during the installation will be related to the installation, such as loss of network connection, etc.
So the idea is to improve (or modify the scope) of `RAUC info` to response at such use case

**Reply by ejoerns** [reply]
We already had some similar requirements, yes. The general idea would be to provide some kind of D-Bus interface that allows user interaction and sending and receiving confirmation requests.
But that did not make it in any proposal or issue afaik
